File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==1)
            return "1";
        string s = countAndSay(n-1), res="";
        char prev=s[0];
        int ct=0;
        for(char c: s){
            if(c==prev)
                ct++;
            else {
                res+=to_string(ct)+to_string(prev-'0');
                ct=1;
                prev=c;
            }
        }
        return res+to_string(ct)+to_string(prev-'0');
    }
};

File: 0038-count-and-say/0038-count-and-say.java
class Solution {
    public String countAndSay(int n) {
        String s = "1";
        while (--n > 0)
            s = solve(s);
        return s;
    }

    private String solve(String s) {
        char prev = s.charAt(0);
        int ct = 1;
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c != prev) {
                sb.append(ct).append(prev);
                prev = c;
                ct = 1;
            } else
                ct++;
        }
        sb.append(ct).append(prev);
        // System.out.println(sb);
        return sb.toString();
    }
}